modules/assim.sequential/R/sda.enkf.R

```diff
@@ -481,51 +515,59 @@ sda.enkf <- function(settings, obs.mean, obs.cov, IC = NULL, Q = NULL){
           
           #analysis
-          ciEnvelope(total.time[(t1:t)],XaCI[,1],XaCI[,2],col=alphapink)
-          lines(total.time[t1:t],Xa,col="black",lty=2,lwd=2)
+          ciEnvelope(strftime(obs.times[t1:t],'%Y'),XaCI[,1],XaCI[,2],col=alphapink)
+          lines(strftime(obs.times[t1:t],'%Y'),Xa,col="black",lty=2,lwd=2)
           
         }
       }
 
     
     ###-------------------------------------------------------------------###
-    ### forecast step -- write restart                                    ###
+    ### forecast step                                                     ###
     ###-------------------------------------------------------------------### 
     if(t < nt){
+      
+      ###-------------------------------------------------------------------###
+      ### load model specific inputs for initial runs                       ###
+      ###-------------------------------------------------------------------### 
+      
+      inputs <- do.call(my.split.inputs,args=list(settings = settings,
+                                                  start.time = format(as.Date(obs.times[t])+1,"%Y/%m/%d"),
+                                                  stop.time = obs.times[t+1]))
+      
+      ###-------------------------------------------------------------------###
+      ### write restart by ensemble                                         ###
+      ###-------------------------------------------------------------------### 
+      
       for(i in 1:nens){
-        if(sample_parameters == TRUE){
-          trait.values <- lapply(ensemble.samples, function(x, n){x[i,]},n = i)
-        } else {
-          trait.values <- ensemble.samples
-        }
         do.call(my.write.restart,
-                args=list(out.dir = outdir, runid = run.id[[i]],
-                          time = total.time[t], settings = settings,
-                          analysis.vec = analysis[i,],
-                          RENAME = TRUE, variables = variables,
-                          sample_parameters = sample_parameters,
-                          trait.values = trait.values,
-                          met = met))
+                args=list(outdir = outdir, runid = run.id[[i]],
+                          start.time = format(as.Date(obs.times[t])+1,"%Y/%m/%d"),
+                          stop.time = obs.times[t+1],
+                          settings = settings,
+                          new.state = new.state[i,],
+                          new.params = new.params[[i]],
+                          inputs = inputs,RENAME=TRUE))
       }
       ###-------------------------------------------------------------------###
       ### Run model                                                         ###
       ###-------------------------------------------------------------------### 
-      print(paste("Running Model for Year",total.time[t]+1))
+      print(paste("Running Model for Year",as.Date(obs.times[t])+1))
```

Comment:
I didn't mean to say you had to think of something else, I just meant that you need to better document what you're doing here. If obs.times is of class Date you should be fine. Also, within write.restart you need to be clear (& document) that for models that can start subdaily that you're starting at 00:00 and ending at 23:59 so that we don't accidentally loose days everytime we stop.